src/neptune/core/operations/operation.py

```diff
@@ -0,0 +1,212 @@
+        return Operation._registry[operation_type].from_dict(data)
+
+
+@dataclass
+class AssignFloat(Operation):
+    value: float
+
+    def accept(self, visitor: "OperationVisitor[Ret]") -> Ret:
+        return visitor.visit_assign_float(self)
+
+    def to_dict(self) -> dict:
+        ret = super().to_dict()
+        ret["value"] = self.value
+        return ret
+
+    @staticmethod
+    def from_dict(data: dict) -> "AssignFloat":
+        return AssignFloat(data["path"], data["value"])
+
+
+@dataclass
+class AssignInt(Operation):
+
+    value: int
+
+    def accept(self, visitor: "OperationVisitor[Ret]") -> Ret:
+        return visitor.visit_assign_int(self)
+
+    def to_dict(self) -> dict:
+        ret = super().to_dict()
+        ret["value"] = self.value
+        return ret
+
+    @staticmethod
+    def from_dict(data: dict) -> "AssignInt":
+        return AssignInt(data["path"], data["value"])
+
+
+@dataclass
+class AssignBool(Operation):
+
+    value: bool
+
+    def accept(self, visitor: "OperationVisitor[Ret]") -> Ret:
+        return visitor.visit_assign_bool(self)
+
+    def to_dict(self) -> dict:
+        ret = super().to_dict()
+        ret["value"] = self.value
+        return ret
+
+    @staticmethod
+    def from_dict(data: dict) -> "AssignBool":
+        return AssignBool(data["path"], data["value"])
+
+
+@dataclass
+class AssignString(Operation):
+
+    value: str
+
+    def accept(self, visitor: "OperationVisitor[Ret]") -> Ret:
+        return visitor.visit_assign_string(self)
+
+    def to_dict(self) -> dict:
+        ret = super().to_dict()
+        ret["value"] = self.value
+        return ret
+
+    @staticmethod
+    def from_dict(data: dict) -> "AssignString":
+        return AssignString(data["path"], data["value"])
+
+
+@dataclass
+class AssignDatetime(Operation):
+
+    value: datetime
+
+    def accept(self, visitor: "OperationVisitor[Ret]") -> Ret:
+        return visitor.visit_assign_datetime(self)
+
+    def to_dict(self) -> dict:
+        ret = super().to_dict()
+        ret["value"] = int(1000 * self.value.timestamp())
+        return ret
+
+    @staticmethod
+    def from_dict(data: dict) -> "AssignDatetime":
+        return AssignDatetime(data["path"], datetime.fromtimestamp(data["value"] / 1000))
+
+
+class LogOperation(Operation, abc.ABC):
+    pass
+
+
+@dataclass
+class LogSeriesValue(Generic[T]):
+
+    value: T
+    step: Optional[float]
+    ts: float
+
+    def to_dict(
+        self,
+        value_serializer: Callable[[T], Any] = lambda x: x,
+    ) -> Dict[str, Union[T, Optional[float], float]]:
+        return {"value": value_serializer(self.value), "step": self.step, "ts": self.ts}
+
+    @staticmethod
+    def from_dict(data: Dict[str, Any], value_deserializer: Callable[[T], Any] = lambda x: x) -> "LogSeriesValue[T]":
+        return LogSeriesValue[T](value_deserializer(data["value"]), data.get("step", None), data["ts"])
```

Comment:
`data.get("step", None)` is the same as `data.get("step")`

Reply:
good point, was just copy-pasting from `internal/operation.py`